x-pack/plugin/identity-provider/docs/en/rest-api/idp-saml-init.asciidoc

```diff
@@ -0,0 +1,122 @@
+[role="xpack"]
+[[idp-saml-init]]
+=== Generate a SAML authentication response message
+++++
+<titleabbrev>Generate SAML response</titleabbrev>
+++++
+Generates a SAML Response message to be sent to a Service Provider
+
+[[idp-saml-init-request]]
+==== {api-request-title}
+
+`POST /_idp/saml/init`
+
+[[idp-saml-init-prereqs]]
+==== {api-prereq-title}
+
+* To use this API, you must have a role that grants the `cluster:admin/idp/saml/init` privilege.
+
+[[idp-saml-init-desc]]
+==== {api-description-title}
+
+This API generates a SAML Response message that should be sent to a Service Provider as part of an
+IDP initiated or SP initiated SAML Single Sign On. This API expects the caller to present
+credentials for the user that the SAML Response will be created for as "Secondary Authentication"
+using the `es-secondary-authorization` HTTP Request header.
+
+The SAML response is returned as an XML String and the caller of the API is responsible to instruct
+the end user's browser to make an HTTP Post request to the Service Provider with the SAML response
+Base64 encoded.
+
+[[idp-saml-init-body]]
+==== {api-request-body-title}
+
+The following parameters can be specified in the body of a POST request:
+
+`entity_id`::
+(Required, string) The SAML entity Id of the service provider that will be the recipient of the SAML Response.
+
+`acs`::
+(Required, string) The assertion consumer service URL of the service provider that will be the recipient of the SAML Response.
+
+`authn_state`::
+(Optional, object) The JSON structure that <<idp-saml-validate>> returns. This should only be
+provided when calling this API as part of an SP initiated Single Sign On.
+
+
+[[idp-saml-init-example]]
+==== {api-examples-title}
+
+The following example generates a SAML Response for an IDP initiated SAML Single Sign On to the Service Provider with entity Id
+`https://sp1.kibana.org` and an assertion consumer service URL that is `https://sp1.kibana.org/saml/acs`
+
+[source, sh]
+--------------------------------------------------------------------
+curl -u idp_admin:idp_admin_pwd <1> -H 'Content-Type: application/json' \
```

Comment:
```suggestion
curl -u idp_admin:idp_admin_pwd <1> \
  -H 'Content-Type: application/json' \
```
I think this will be neater if the footnotes are always at the end of line (but that's a personal view)